### PR TITLE
Move workflow trigger filter to job

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Move workflow trigger filter to job, because if workflow is not even triggered the required checks for merging won't even activate and pr would be blocked for merging:

image

### CODE REVIEW

File: .github/workflows/social-connect.yml [hunk 1]
```diff
@@ -186,6 +182,12 @@ jobs:
     container:
       image: us-west1-docker.pkg.dev/devopsre/actions-runner-controller/celo-monorepo:node18
     needs: install-dependencies
+    if: |
+      contains(needs.install-dependencies.outputs.all_modified_files, 'packages') ||
+      contains(needs.install-dependencies.outputs.all_modified_files, '.github') ||
+      contains(needs.install-dependencies.outputs.all_modified_files, '.changeset') ||
+      contains(needs.install-dependencies.outputs.all_modified_files, 'packafe.json') ||
```
Comment (aaronmgdr) [inline .github/workflows/social-connect.yml:189]: typo `packafe.json` => `package.json`

**Review by aaronmgdr — CHANGES_REQUESTED** [review]
fix typo

File: .github/workflows/social-connect.yml (hunk 1 above)
Comment (aaronmgdr) [reply .github/workflows/social-connect.yml:189]: im interpreting this as "only run ci checks when a file matching one of these descriptions is changed" 

Im a bit worried about missing something. I think an explicit "dont run if only X file" is changed would feel safer. if not possible we should at least add 

* dockerfiles
* docs/examples

Comment (jcortejoso) [reply .github/workflows/social-connect.yml]: Updated! I think we can run the general-test always.

**Review by aaronmgdr — APPROVED** [review]

**Review by soloseng — APPROVED** [review]
🚀

**Pull request merged**